src/pages/ExamsPage/components/ExamSelection.jsx

```diff
@@ -22,11 +21,17 @@ const ExamSelection = ({ exams, onSelect }) => {
         onSubmit={() => {}}
       />,
     ];
-    return menuItems.concat(exams.filter(exam => (
+    const examsMatchSearch = exams.filter(exam => (
       exam.name.toLowerCase().includes(searchText.toLowerCase())
     )).map(
       exam => <MenuItem key={exam.id} onClick={() => onSelect(exam.id)}>{exam.name}</MenuItem>,
-    ));
+    );
+    const examsNotMatchSearch = exams.filter(exam => !(
+      exam.name.toLowerCase().includes(searchText.toLowerCase())
+    )).map(
+      exam => <MenuItem disabled>{exam.name}</MenuItem>,
```

Comment:
You know, I'm not sure! I was thinking so though, since like you said, it is an array. I just kept the filter, map, as it was before. Would there be a reason to change it? I also figured that this would help preserve the order?

Reply:
Sorry, I was trying to ask if we also need to add a `key` here, like you did at line 27. I think we do need the `key` prop here. I see this warning in the console.

```
  console.error
    Warning: Each child in a list should have a unique "key" prop.

    Check the render method of `ExamSelection`. See https://reactjs.org/link/warning-keys for more information.
        in MenuItem (created by ExamSelection)
        in ExamSelection

      37 |   return (
      38 |     <div data-testid="exam_selection">
    > 39 |       <SelectMenu
         |       ^
      40 |         defaultMessage={formatMessage(messages.examSelectDropdownLabel)}
      41 |       >
      42 |         { getMenuItems() }
```

This warning goes away if you add a `key` prop here.

Reply:
Oh haha. Sorry I misunderstood. That makes sense. Added the key prop.